USAGE.md

```diff
@@ -0,0 +1,274 @@
+# ECS Tooling Usage
+
+In addition to the published schema and generated artifacts, the ECS repo also contains tools to generate artifacts based on the current published and custom schemas.
+
+**NOTE** - These tools and their functionality are considered experimental.
+
+## Table of Contents
+
+- [Terminology](#terminology)
+- [Setup and Install](#setup-and-install)
+  * [Prerequisites](#prerequisites)
+    + [Clone from GitHub](#clone-from-github)
+    + [Option 1: Install dependencies via make (recommended)](#option-1-install-dependencies-via-make-recommended)
+    + [Option 2: Install dependencies via pip](#option-2-install-dependencies-via-pip)
+- [Usage](#usage)
+  * [Getting Started - Generating Artifacts](#getting-started---generating-artifacts)
+  * [Generator Options](#generator-options)
+    + [Out](#out)
+    + [Include](#include)
+    + [Subset](#subset)
+    + [Ref](#ref)
+    + [Mapping & Template Settings](#mapping--template-settings)
+    + [Intermediate-Only](#intermediate-only)
+
+## Terminology
+
+| Term | Definition |
+| ---- | ---------- |
+| ECS | Elastic Common Schema. For the purposes of this guide, ECS may refer to either the schema itself or the repo/tooling used to maintain the schema |
+| artifacts | Various kinds of files or programs that can be generated based on ECS |
+| field set | Groups of related fields in ECS |
+| schema | Another term for a group of related fields in ECS. Used interchangeably with field set |
+| schema definition | The markup used to define a schema in ECS |
+| attributes | The properties of a field or field set that are used to define that field or field set in a schema definition |
+
+## Setup and Install
+
+### Prerequisites
+
+* [Python 3.6+](https://www.python.org/)
+* [make](https://www.gnu.org/software/make/)
+* [pip](https://pypi.org/project/pip/)
+* [git](https://git-scm.com/)
+
+#### Clone from GitHub
+
+The recommended way to download the ECS repo is `git clone`:
+
+```
+$ git clone https://github.com/elastic/ecs
+$ cd ecs
+```
+
+Prior to installing dependencies or running the tools, it's recommended to check out the `git` branch for the ECS version being targeted.
+
+**Example**: For ECS `1.5.0`:
+
+```
+$ git checkout v1.5.0
+```
```

Comment:
Now that we have `--ref` we don't need this step anymore :-)

Reply:
Great point, we should definitely promote `--ref` instead. 

I ran up against an issue (as I was going to update the docs 😄 ) trying to run the generator with `--ref` (#892), so I may hold off removing this until we squash that bug.